code/modules/mod/modules/modules.dm

```diff
@@ -0,0 +1,1770 @@
+	RegisterSignal(mod.wearer, COMSIG_HUMAN_CHECK_SHIELDS, .proc/shield_reaction)
+
+/obj/item/mod/module/energy_shield/on_suit_deactivation()
+	var/datum/component/shielded/shield = mod.GetComponent(/datum/component/shielded)
+	charges = shield.current_charges
+	qdel(shield)
+	UnregisterSignal(mod.wearer, COMSIG_HUMAN_CHECK_SHIELDS)
+
+/obj/item/mod/module/energy_shield/proc/shield_reaction(mob/living/carbon/human/owner, atom/movable/hitby, damage = 0, attack_text = "the attack", attack_type = MELEE_ATTACK, armour_penetration = 0)
+	if(SEND_SIGNAL(mod, COMSIG_ITEM_HIT_REACT, owner, hitby, attack_text, 0, damage, attack_type) & COMPONENT_HIT_REACTION_BLOCK)
+		drain_power(use_power_cost)
+		return SHIELD_BLOCK
+	return NONE
+
+/obj/item/mod/module/energy_shield/wizard
+	name = "MOD battlemage shield module"
+	desc = "The caster wielding this spell gains a visible barrier around them, channeling arcane power through \
+		specialized runes engraved onto the surface of the suit to generate a wall of force. \
+		This shield can perfectly nullify attacks ranging from high-caliber rifles to magic missiles, \
+		though can also be drained by more mundane attacks. It will not protect the caster from social ridicule."
+	icon_state = "battlemage_shield"
+	idle_power_cost = DEFAULT_CELL_DRAIN * 0 //magic
+	use_power_cost = DEFAULT_CELL_DRAIN * 0 //magic too
+	max_charges = 15
+	recharge_start_delay = 0 SECONDS
+	charge_recovery = 8
+	recharge_path = /obj/item/wizard_armour_charge
+
+/obj/item/mod/module/plasma_stabilizer
+	name = "MOD plasma stabilizer module"
+	desc = "This system essentially forms an atmosphere of its' own inside the suit, \
+		safely ejecting oxygen from the inside and allowing the wearer, a plasmaman, \
+		to have their internal plasma circulate around them somewhat like a sauna. \
+		This prevents them from self-igniting, and leads to greater comfort overall. \
+		The purple glass of the visor seems to be constructed for nostalgic purposes."
+	icon_state = "plasma_stabilizer"
+	complexity = 1
+	idle_power_cost = DEFAULT_CELL_DRAIN * 0.3
+	incompatible_modules = list(/obj/item/mod/module/plasma_stabilizer)
+	overlay_state_inactive = "module_plasma"
+
+/obj/item/mod/module/plasma_stabilizer/on_equip()
+	ADD_TRAIT(mod.wearer, TRAIT_NOSELFIGNITION, MOD_TRAIT)
+
+/obj/item/mod/module/plasma_stabilizer/on_unequip()
+	REMOVE_TRAIT(mod.wearer, TRAIT_NOSELFIGNITION, MOD_TRAIT)
+
+/obj/item/mod/module/anti_magic
+	name = "MOD magic nullifier module"
+	desc = "A series of obsidian rods installed into critical points around the suit, \
+		vibrated at a certain low frequency to enable them to resonate. \
+		This creates a low-range, yet strong, magic nullification field around the user, \
+		aided by a full replacement of the suit's normal coolant with holy water. \
+		Spells will spall right off this field, though it'll do nothing to help others believe you about all this."
```

Comment:
Is this supposed to say `spall`? This feels like it could be a reference I don't get

Reply:
yes